app/scripts/migrations/048.js

```diff
@@ -20,14 +22,23 @@ export default {
   },
 }
 
+const inTest = process.env.IN_TEST === 'true'
+
 function transformState (state = {}) {
   // 1. Delete NetworkController.settings
   delete state.NetworkController?.settings
 
-  // 2. Delete NetworkController.provider or rename rpcTarget key
+  // 2. Migrate NetworkController.provider to Rinkeby or rename rpcTarget key
   const providerType = state.NetworkController?.provider?.type
-  if (providerType === 'rpc' || providerType === 'localhost') {
-    delete state.NetworkController.provider
+  if (!inTest && (providerType === 'rpc' || providerType === 'localhost')) {
```

Comment:
This will bump all users on custom RPC endpoints to rinkeby, even those with correctly specified `chainId`s. We could reduce the impact of this change by only bumping users from networks that require manual changes.

This would also remove the need for this `inTest` variable, as the localhost RPC endpoint is correctly specified, so it wouldn't need to be bumped.

Reply:
We'd want to use exactly the same check as we do at runtime: https://github.com/MetaMask/metamask-extension/blob/develop/ui/app/components/app/dropdowns/network-dropdown.js#L109

Reply:
7321d58